modules/shared/src/main/scala/io/constellationnetwork/schema/swap.scala

```diff
@@ -36,8 +94,51 @@ object swap {
     parent: AllowSpendReference,
     lastValidOrdinal: SnapshotOrdinal,
     approvers: List[Address]
+  ) extends SwapTransaction {
+    val ordinal: AllowSpendOrdinal = parent.ordinal.next
+  }
+
+  object AllowSpend {
+    implicit object OrderingInstance extends OrderBasedOrdering[AllowSpend]
+  }
+
+  @derive(encoder)
+  case class AllowSpendView(
+    transaction: AllowSpend,
+    hash: Hash,
+    status: AllowSpendStatus
   )
 
+  @derive(eqv, show)
+  sealed trait AllowSpendStatus extends EnumEntry
+
+  object AllowSpendStatus extends Enum[AllowSpendStatus] with AllowSpendStatusCodecs {
+    val values = findValues
+
+    case object Waiting extends AllowSpendStatus
+  }
+
+  trait AllowSpendStatusCodecs {
+    implicit val encode: Encoder[AllowSpendStatus] = Encoder.encodeString.contramap[AllowSpendStatus](_.entryName)
+    implicit val decode: Decoder[AllowSpendStatus] =
+      Decoder.decodeString.emapTry(s => Try(AllowSpendStatus.withName(s)))
+  }
+
+  @derive(decoder, encoder, show)
+  case class SwapBlock(
+    roundId: RoundId,
+    transactions: NonEmptySet[Signed[SwapTransaction]]
```

Comment:
Perhaps I am missing something but who would generate the proofs associated with the `SwapTransaction`? My understanding is that these transaction should be automatically generated by the metagraph when there are matching `AllowSpend` transactions within the order book maintained by the AMM metagraph.

Reply:
Good question, but I don't know the answer. I was following our old meetings' notes. Maybe
@ryle-goehausen @AlexBrandes?

Reply:
Hey, I would reference the Currency Swapping notion doc for latest information. As far as I can tell this is just a naming issue with the trait SwapTransaction. We've eliminated MetagraphAllowSpend in favor of SpendTransaction (generated via metagraph snapshot embedded). So I think the user defined AllowSpend is fine to use with Signed[] (since it's wallet produced.) If we have the final SpendTransaction it won't conform to this trait though since the signatures are associated with the metagraph snapshot.

SwapTransaction might be more appropriate name internal to the AMM logic, which I think is what James is referring to just to clarify.

Reply:
@ryle-goehausen do I understand correctly that AllowSpend Transaction should be Signed by individuals and SpendTransaction won't be ever signed?

Reply:
yes, that was my understanding. The `SwapTransaction` (or whichever transaction actually changes balances) is meant to be verified via an inclusion proof within a `CurrencySnapshot`, while an `AllowSpend` is a user generated "authorization" that can be verified by a signature proof.

Reply:
Hmm, it complicates things a little bit as we talked about making those transactions generic, meaning they need to follow the same format/flow. Looks like there is no point to merge allow spends and spend txs under the same trait. 

The other solution would be to sign it by all the peers under the consensus but it doesn't make much sense. It may work like reward transactions (so no signature at all)